src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -466,15 +529,53 @@ private static ImmutableArray<Diagnostic> FilterDiagnosticsSuppressedInSource(Im
 #endif
 
                 var diagnostic = SuppressMessageAttributeState.ApplySourceSuppressions(diagnostics[i], compilation);
-                if (reportSuppressedDiagnostics || !diagnostic.IsSuppressed)
+                if (!reportSuppressedDiagnostics && diagnostic.IsSuppressed)
                 {
-                    builder.Add(diagnostic);
+                    // Diagnostic suppressed in source.
+                    continue;
                 }
+
+                builder.Add(diagnostic);
             }
 
             return builder.ToImmutable();
         }
 
+        private bool IsInGeneratedCode(Location location, Compilation compilation)
+        {
+            if (!location.IsInSource)
+            {
+                return false;
+            }
+
+            if (IsGeneratedCode(location.SourceTree))
+            {
+                return true;
+            }
+
+            if (_generatedCodeAttribute != null)
+            {
+                var model = compilation.GetSemanticModel(location.SourceTree);
+                for (var node = location.SourceTree.GetRoot().FindNode(location.SourceSpan, getInnermostNodeForTie: true);
```

Comment:
Changed to pass in the cancellation token.
FindToken extension method on Location is in Workspaces layer - not sure there are any other cases where we would want to go from location to syntax in compiler layer.